lib/ansible/plugins/connection/ssh.py

```
@@ -268,6 +268,16 @@
         vars:
           - name: ansible_ssh_use_tty
             version_added: '2.7'
+        yaml: {key: connection.usetty}
+      pkcs11_provider:
```

This should probably change in one of two ways to allow PKCS11 support to be enabled on a per-host basis:
a) add `vars: { name=ansible_ssh_pkcs11_provider }` to make the value settable in inventory at the host/group level 
b) add a second config value with vars/ini/env/yaml values like `pkcs11_enabled`

While the actual provider configuration makes sense as a global config/env option (since the provider itself is unlikely to vary by host), there should probably be a separate config+env+var option to turn it on for a given host. Otherwise it's an all-or-none proposition for using it, where in complex environments, you may need both in the same Ansible run. Two options is a bit more hassle, but would allow the global configuration of the provider while hosts could still opt in/out individually (or by group). 

This could also allow you to add a warning that the token PIN is required with PKCS11 (rather than silently failing to auth as it would now by bypassing the PKCS11 path entirely).

---

I went with option a.  If you specify the pkcs11 provider then pkcs11 is enabled, if you don't specify the provider then its disabled. I could detect if pkcs11 provider was used but no password was provided, I could print an error that a password/pin is required.  I am trying to avoid adding too many variables, it seems that if you specify a pkcs11_provider then its safe to assume your using pkcs11.  Let me know what you think.